app/Repositories/MessageRepository.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Gladiator\Repositories;
+
+use Gladiator\Models\Message;
+
+class MessageRepository
+{
+    /**
+     * Create a new message.
+     *
+     * @param  \Gladiator\Models\Contest $contest
+     * @param  array $data
+     * @return \Gladiator\Models\Message
+     */
+    public function create($contest, $data)
+    {
+        $message = new Message($data);
+
+        return $contest->messages()->save($message);
```

Comment:
@angaither cleaner! 

Although I actually tried to do something like what we're doing in the `ContestController@store()` method, where we do:

```php
...
$contest->waitingRoom->fill($request->only(['signup_start_date', 'signup_end_date']))->save();
```

So I tried doing:

```php
$contest->messages->fill($data)->save();
```

But I kept getting an error that the `fill()` method didn't exist. Not sure why, but my suspicion is because the relationship is a `hasMany` and thus continues on with an expected collection of messages, as opposed to the `waitingRoom` example where that is a `hasOne`... but otherwise not sure. Do you or maybe @DFurnes know why it didn't think the method existed in this scenario?